pkg/webhook/admission/webhook.go

```diff
@@ -84,23 +80,30 @@ type Webhook struct {
 }
 
 func (w *Webhook) setDefaults() {
-	if len(w.Name) == 0 {
-		w.Name = "admissionwebhook.k8s.io"
-	}
 	if len(w.Path) == 0 {
+		if len(w.Rules) == 0 || len(w.Rules[0].Resources) == 0 {
+			// can't do defaulting, skip it.
+			return
+		}
 		if w.Type == types.WebhookTypeMutating {
-			w.Path = "/mutatingwebhook"
+			w.Path = "/mutate-" + w.Rules[0].Resources[0]
 		} else if w.Type == types.WebhookTypeValidating {
-			w.Path = "/validatingwebhook"
+			w.Path = "/validate-" + w.Rules[0].Resources[0]
 		}
 	}
+	if len(w.Name) == 0 {
+		reg := regexp.MustCompile("[^a-zA-Z0-9]+")
+		processedPath := strings.ToLower(reg.ReplaceAllString(w.Path, ""))
+		w.Name = processedPath + ".k8s.io"
```

Comment:
is `.k8s.io` always the right suffix?  that seems wrong...

Reply:
In kubebuilder scaffolding, kubebuilder should use the domain name. i.e. `builder.Name("myfeature.mycorp.com")`
But in the library defaulting, there is no such information. IMO using `k8s.io` as default in the library is fine.

Reply:
k8s.io is for official kubernetes projects.  We should not encourage people to be misusing that suffix.